NitroxClient/GameLogic/Bases/Spawning/BasePiece/BasePieceSpawnPrioritizer.cs

```diff
@@ -77,5 +81,54 @@ private int ComputeBasePiecePriority(BasePieceData basePiece)
 
             return piecesToPriority.TryGetValue(techType, out int position) ? position : int.MaxValue;
         }
+
+        private List<BasePieceData> OrderByHeight(IEnumerable<BasePieceData> orderedList, TechType techType)
+        {
+            int i = 0;
+            int minIndex = -1;
+            int maxIndex = -1;
+            foreach (BasePieceData basePieceData in orderedList)
```

Comment:
If you need an idex here would a traditional for loop not work just as well? Saying that I'm not sure on how c# will optimise the loop here on access to the list

Reply:
We can probably let it like this for now and when we have a better solution we'll correct it. Also remember this is only happening one time at the beginning